### ISSUE TITLE
Traefik drop request when nginx trigger configure reload

### ISSUE THREAD

**Issue by victormakmo** [containous/traefik#2210]
### Do you want to request a *feature* or report a *bug*?

bug

### What did you do?

Use traefik and nginx proxy similar zero downtime blue green deployment.

Client (AB test) -> Traefik -> nginx proxy -> whoami01, whoami02
                                                                  
1. Deploy traefik and nginx proxy
2. Deploy two service whoami01, whoami02
3. Perform AB test
4. while performing AB test, modify nginx upstream point to whoami02, reload nginx config
5. wait until AB test finish

### What did you expect to see?
I expect no packet loss

### What did you see instead?
packet loss

### Output of `traefik version`: (_What version of Traefik are you using?_)
v1.3.5 / raclette

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

**Docker version**
```
Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
```

**Linux version:**
`CentOS Linux release 7.1.1503 (Core) kernel 4.12.5-1.el7.elrepo.x86_64`

**Traefik docker compose**:

```yml
version: '3'
services:
  app:
    image: 'traefik:1.3.5'
    command: >-
      --web --docker --docker.domain=mylab.com --web.statistics --logLevel=DEBUG
      --docker.swarmmode
    networks:
      - tst_network
    ports:
      - '80:80'
      - '8088:8080'
    volumes:
      - '/var/run/docker.sock:/var/run/docker.sock'
    deploy:
      mode: global
      placement:
        constraints:
          - node.role == manager
networks:
  tst_network:
    external: true
```

**nginx docker compose:**
```yml
version: '3'
services:
  app:
    image: nginx
    volumes:
        - "./etc/nginx/default.conf:/etc/nginx/conf.d/default.conf"
    deploy:
      labels:
        - traefik.port=80
        - traefik.backend=tst_whoaminginx_app
        - 'traefik.frontend.rule=Host:whoami.mylab.com;'
    networks:
      - tst_network
networks:
  tst_network:
    external: true
```

nginx configure

```
worker_processes  1;
user root root;
events {
   use epoll;
   worker_connections  1024;
}
http {

        resolver 127.0.0.11;

        upstream service {
                server whoami01:8000;
        }

        server {
                listen       80;
                access_log  /var/log/nginx/access.log;
                error_log  /var/log/nginx/error.log;
                root   /usr/share/nginx/html;
                index  index.html index.htm;
                location / {
                        proxy_pass  http://service;
                        proxy_set_header Host $host;
                        proxy_set_header X-Real-IP $remote_addr;
                        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                        proxy_connect_timeout 90;
                        proxy_send_timeout 90;
                        proxy_read_timeout 90;
                }

        }
}

```
**whoami01 docker compose:**
```yml
version: '3'
services:
  app:
    image: jwilder/whoami
    networks:
      tst_network:
        aliases:
          - whoami01
networks:
  tst_network:
    external: true
```

**whoami02 docker compose:**
```yml
version: '3'
services:
  app:
    image: jwilder/whoami
    networks:
      tst_network:
        aliases:
          - whoami02
networks:
  tst_network:
    external: true

```

I got 75 failed requests after the AB test.
`docker run edseymour/ab -n 100000 -c 100 http://whoami.mylab.com/`

**AB test report:**

```
command line provided overrides ENVs
**********************************
Test starting Tue Oct  3 14:47:37 UTC 2017
**********************************
Running now: ab -n 100000 -c 100 http://whoami.mylab.com/
This is ApacheBench, Version 2.3 <$Revision: 1430300 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking whoami.mylab.com (be patient)
Completed 10000 requests
Completed 20000 requests
Completed 30000 requests
Completed 40000 requests
Completed 50000 requests
Completed 60000 requests
Completed 70000 requests
Completed 80000 requests
Completed 90000 requests
Completed 100000 requests
Finished 100000 requests


Server Software:        nginx/1.12.1
Server Hostname:        whoami.mylab.com
Server Port:            80

Document Path:          /
Document Length:        17 bytes

Concurrency Level:      100
Time taken for tests:   393.008 seconds
Complete requests:      100000
Failed requests:        75
   (Connect: 0, Receive: 0, Length: 75, Exceptions: 0)
Write errors:           0
Non-2xx responses:      75
Total transferred:      15599395 bytes
HTML transferred:       1699960 bytes
Requests per second:    254.45 [#/sec] (mean)
Time per request:       393.008 [ms] (mean)
Time per request:       3.930 [ms] (mean, across all concurrent requests)
Transfer rate:          38.76 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0  179 680.5      5    7052
Processing:     2  213 418.1     72    7178
Waiting:        0  212 418.2     72    7178
Total:          2  392 882.2     84   10093

Percentage of the requests served within a certain time (ms)
  50%     84
  66%     99
  75%    116
  80%    146
  90%   1033
  95%   2048
  98%   3013
  99%   4073
 100%  10093 (longest request)
**********************************
Test finished Tue Oct  3 14:54:10 UTC 2017
**********************************
```

**Comment by Juliens** [comment]
May be what you want is https://docs.traefik.io/configuration/commons/#retry-configuration ?

**Comment by victormakmo** [comment]
@Juliens 

After tried your solution, i still got some failure requests.

```
**********************************
Test starting Fri Oct  6 09:53:24 UTC 2017
**********************************
Running now: ab -n 20000 -c 50 http://whoami.mylab.com/
This is ApacheBench, Version 2.3 <$Revision: 1430300 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking whoami.mylab.com (be patient)
Completed 2000 requests
Completed 4000 requests
Completed 6000 requests
Completed 8000 requests
Completed 10000 requests
Completed 12000 requests
Completed 14000 requests
Completed 16000 requests
Completed 18000 requests
Completed 20000 requests
Finished 20000 requests


Server Software:        nginx/1.12.1
Server Hostname:        whoami.mylab.com
Server Port:            80

Document Path:          /
Document Length:        17 bytes

Concurrency Level:      50
Time taken for tests:   74.100 seconds
Complete requests:      20000
Failed requests:        12
   (Connect: 0, Receive: 0, Length: 12, Exceptions: 0)
Write errors:           0
Non-2xx responses:      12
Total transferred:      3119972 bytes
HTML transferred:       340028 bytes
Requests per second:    269.90 [#/sec] (mean)
Time per request:       185.251 [ms] (mean)
Time per request:       3.705 [ms] (mean, across all concurrent requests)
Transfer rate:          41.12 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    6   7.4      4     109
Processing:     2  178 339.0     43    1276
Waiting:        1  177 339.2     43    1276
Total:          3  184 337.3     50    1279

Percentage of the requests served within a certain time (ms)
  50%     50
  66%     60
  75%     69
  80%     80
  90%   1023
  95%   1025
  98%   1028
  99%   1067
 100%   1279 (longest request)
**********************************
Test finished Fri Oct  6 09:54:38 UTC 2017
**********************************

```

Below is my updated configure:

traefik:
```
version: '3'
services:
  app:
    image: 'traefik:1.3.5'
    command: >-
      --web --docker --docker.domain=mylab.com --web.statistics --logLevel=DEBUG
      --docker.swarmmode
    networks:
      - tst_network
    ports:
      - '80:80'
      - '8088:8080'
    volumes:
      - '/var/run/docker.sock:/var/run/docker.sock'
      - '/tmp/traefik.toml:/etc/traefik/traefik.toml'
    deploy:
      mode: global
      placement:
        constraints:
          - node.labels.type == manager
networks:
  tst_network:
    external: true
```

trarfik.toml
```
################################################################
# Global configuration
################################################################
# Log level
logLevel = "DEBUG"
#
# Entrypoints to be used by frontends that do not specify any entrypoint.
# Each frontend can specify its own entrypoints.
defaultEntryPoints = ["http", "https"]
#
# Enable retry sending request if network error
[retry]
# Number of attempts
#
# Optional
# Default: (number servers in backend) -1
#
attempts = 3

################################################################
# Web configuration backend
################################################################

# Enable web configuration backend
[web]

# Web administration port
address = ":8080"

################################################################
# Docker configuration backend
################################################################

# Enable Docker configuration backend
[docker]

# Docker server endpoint. Can be a tcp or a unix socket endpoint.
endpoint = "unix:///var/run/docker.sock"

# Default domain used.
# Can be overridden by setting the "traefik.domain" label on a container.
domain = "mylab.com"

# Enable watch docker changes
watch = true

# Optional
# Default: (number servers in backend) -1
#
attempts = 3

################################################################
# Web configuration backend
################################################################

# Enable web configuration backend
[web]

# Web administration port
address = ":8080"

################################################################
# Docker configuration backend
################################################################

# Enable Docker configuration backend
[docker]

# Docker server endpoint. Can be a tcp or a unix socket endpoint.
endpoint = "unix:///var/run/docker.sock"

# Default domain used.
# Can be overridden by setting the "traefik.domain" label on a container.
domain = "mylab.com"

# Enable watch docker changes
watch = true
```

**Comment by victormakmo** [comment]
may i know any update on this issue?

**Comment by emilevauge** [comment]
@victormakmo What is the use case of chaining Traefik + Nginx? Can you test this without Nginx instead ?

**Comment by victormakmo** [comment]
@emilevauge the use case is i want to perform zero downtime rolling update or blue/green deployment.  However, as mentioned in https://github.com/containous/traefik/issues/1480 and https://github.com/moby/moby/pull/32421, traefik got some packets lost during scaling and rolling update period. I try to workaround this case is to combined traefik and nginx together since nginx reload process gave us no packet loss during the update period.

**Comment by ldez** [comment]
Closed in favor of #3035
